crest/Assets/Crest/Crest/Scripts/OceanValidation.cs

```diff
@@ -95,11 +103,37 @@ public static void RunValidation(OceanRenderer ocean)
             var inputs = FindObjectsOfType<RegisterLodDataInputBase>();
             foreach (var input in inputs)
             {
-                input.Validate(ocean);
+                input.Validate(ocean, ValidatedHelper.DebugLog);
             }
 
             Debug.Log("Validation complete!", ocean);
         }
     }
+
+    // Holds the shared list for messages
+    public static class ValidatedHelper
+    {
+        public delegate void ShowMessage(string message, MessageType type, Object @object = null);
+
+        public static void DebugLog(string message, MessageType type, Object @object = null)
+        {
+            message += " Click this error to highlight the problem object.";
+            switch (type)
+            {
+                // NOTE: this is incomplete
+                case MessageType.Warning: Debug.LogWarning(message, @object); break;
```

Comment:
https://github.com/Unity-Technologies/UnityCsReference/blob/9034442437e6b5efe28c51d02e978a96a3ce5439/Runtime/Export/Logging/Logger.cs

I would probably use LogFormat here. Avoids the switch statement.

Reply:
Unfortunately, LogType doesn't map nicely to MessageType. Furthermore, LogType is used by the ILogger interface which is exposed under `Debug.unityLogger` rather than the normal methods. I'm not sure it is a good idea to use it as it might bypass custom consoles:
> Logs message to the Unity Console using default logger.

Not sure if that means anything or not. I don't have a custom console asset to test against.

I have created our own enum since we won't use all the states for either of these enums.